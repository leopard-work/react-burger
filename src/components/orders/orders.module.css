.page {
    height: 924px;
}
.wrapper {
    overflow-y: scroll;
    scroll-behavior: smooth;
    height: 100%;
}
.wrapper::-webkit-scrollbar {
    width: 8px;
}
.wrapper::-webkit-scrollbar-track {
    background: #2F2F37;
}
.wrapper::-webkit-scrollbar-thumb {
    background-color: #8585AD;
}
.item {
    border-radius: 40px;
    background: #1C1C21;
    cursor: pointer;
}
.top {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
    align-content: center;
}
.bottom {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
    align-content: center;
}
.params {
    display: flex;
    flex-direction: row-reverse;
    flex-wrap: wrap;
    justify-content: flex-start;
    align-items: center;
    align-content: center;
}
.param {
    width: 64px;
    height: 64px;
    border: solid 2px greenyellow;
    border-radius: 50%;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    align-content: center;
    margin-left: -15px;
    background-color: #1c1c21;
}
.param:last-child {
    margin-left: 0px;
}
.price {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: flex-start;
    align-items: center;
    align-content: center;
}

