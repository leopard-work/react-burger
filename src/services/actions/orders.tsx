export const ORDERS_CONNECT: "ORDERS_CONNECT" = "ORDERS_CONNECT";
export const ORDERS_DISCONNECT: "ORDERS_DISCONNECT" = "ORDERS_DISCONNECT";
export const ORDERS_WS_CONNECTING: "ORDERS_WS_CONNECTING" =
  "ORDERS_WS_CONNECTING";
export const ORDERS_WS_OPEN: "ORDERS_WS_OPEN" = "ORDERS_WS_OPEN";
export const ORDERS_WS_CLOSE: "ORDERS_WS_CLOSE" = "ORDERS_WS_CLOSE";
export const ORDERS_WS_ERROR: "ORDERS_WS_ERROR" = "ORDERS_WS_ERROR";
export const ORDERS_WS_MESSAGE: "ORDERS_WS_MESSAGE" = "ORDERS_WS_MESSAGE";

export const OPEN_ORDER_ITEM: "OPEN_ORDER_ITEM" = "OPEN_ORDER_ITEM";
export const CLOSE_ORDER_ITEM: "CLOSE_ORDER_ITEM" = "CLOSE_ORDER_ITEM";

interface ORDERS_CONNECT_ACTION {
  readonly type: typeof ORDERS_CONNECT;
}
interface ORDERS_DISCONNECT_ACTION {
  readonly type: typeof ORDERS_DISCONNECT;
}
interface ORDERS_WS_CONNECTING_ACTION {
  readonly type: typeof ORDERS_WS_CONNECTING;
}
interface ORDERS_WS_OPEN_ACTION {
  readonly type: typeof ORDERS_WS_OPEN;
}
interface ORDERS_WS_CLOSE_ACTION {
  readonly type: typeof ORDERS_WS_CLOSE;
}
interface ORDERS_WS_ERROR_ACTION {
  readonly type: typeof ORDERS_WS_ERROR;
  payload: {
    error: string;
  };
}
interface ORDERS_WS_MESSAGE_ACTION {
  readonly type: typeof ORDERS_WS_MESSAGE;
  payload: {
    data: any;
  };
}
interface OPEN_ORDER_ITEM {
  readonly type: typeof OPEN_ORDER_ITEM;
}

interface CLOSE_ORDER_ITEM {
  readonly type: typeof CLOSE_ORDER_ITEM;
}

export type ORDERS_ACTIONS =
  | ORDERS_CONNECT_ACTION
  | ORDERS_DISCONNECT_ACTION
  | ORDERS_WS_CONNECTING_ACTION
  | ORDERS_WS_OPEN_ACTION
  | ORDERS_WS_CLOSE_ACTION
  | ORDERS_WS_ERROR_ACTION
  | ORDERS_WS_MESSAGE_ACTION
  | OPEN_ORDER_ITEM
  | CLOSE_ORDER_ITEM;
