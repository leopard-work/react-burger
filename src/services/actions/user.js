import {loginUserAPI, registerUserAPI} from "../../utils/api";

export const GET_REGISTER_REQUEST = 'GET_REGISTER_REQUEST';
export const GET_REGISTER_SUCCESS = 'GET_REGISTER_SUCCESS';
export const GET_REGISTER_FAILED = 'GET_REGISTER_FAILED';
export const GET_LOGIN_REQUEST = 'GET_LOGIN_REQUEST';
export const GET_LOGIN_SUCCESS = 'GET_LOGIN_SUCCESS';
export const GET_LOGIN_FAILED = 'GET_LOGIN_FAILED';

export function registerUser(body) {
    return function(dispatch) {
        dispatch({type: GET_REGISTER_REQUEST});
        registerUserAPI(body).then((data) => {
            if (data.success) dispatch({type: GET_REGISTER_SUCCESS, data: data});
            else dispatch({type: GET_REGISTER_FAILED});
        }).catch(() => {
            dispatch({type: GET_REGISTER_FAILED});
        })
    };
}

export function loginUser(body) {
    return function(dispatch) {
        dispatch({type: GET_LOGIN_REQUEST});
        loginUserAPI(body).then((data) => {
            if (data.success) dispatch({type: GET_LOGIN_SUCCESS, data: data});
            else dispatch({type: GET_LOGIN_FAILED});
        }).catch(() => {
            dispatch({type: GET_LOGIN_FAILED});
        })
    };
}