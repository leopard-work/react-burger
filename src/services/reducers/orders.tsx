import {
  ORDERS_CONNECT,
  ORDERS_DISCONNECT,
  ORDERS_WS_CONNECTING,
  ORDERS_WS_OPEN,
  ORDERS_WS_CLOSE,
  ORDERS_WS_ERROR,
  ORDERS_WS_MESSAGE,
  ORDERS_ACTIONS,
  OPEN_ORDER_ITEM,
  CLOSE_ORDER_ITEM,
} from "../actions/orders";
import { OrderItemProps } from "../../utils/types";

type ordersState = {
  ordersData: {
    success: boolean;
    orders: any;
    total: number;
    totalToday: number;
  };
  status: string;
  connect: boolean;
  error: string;
  viewFullOrder: OrderItemProps;
  viewFullOrderModalOpen: boolean;
};

export const initialState: ordersState = {
  ordersData: {
    success: false,
    orders: [],
    total: 0,
    totalToday: 0,
  },
  status: "",
  connect: false,
  error: "",
  viewFullOrder: {
    _id: "",
    ingredients: [],
    status: "",
    name: "",
    createdAt: "",
    updatedAt: "",
    number: 0,
  },
  viewFullOrderModalOpen: false,
};

export const ordersReducer = (state = initialState, action: ORDERS_ACTIONS) => {
  switch (action.type) {
    case ORDERS_CONNECT: {
      return {
        ...state,
        status: "connect...",
      };
    }
    case ORDERS_DISCONNECT: {
      return initialState;
    }
    case ORDERS_WS_CONNECTING: {
      return {
        ...state,
        status: "connecting",
        connect: true,
      };
    }
    case ORDERS_WS_OPEN: {
      return {
        ...state,
        status: "open",
      };
    }
    case ORDERS_WS_MESSAGE: {
      return {
        ...state,
        status: "message",
        ordersData: action.data,
      };
    }
    case ORDERS_WS_CLOSE: {
      return {
        ...state,
        status: "close",
        connect: false,
      };
    }
    case ORDERS_WS_ERROR: {
      return {
        ...state,
        status: "error",
        error: action.error,
        connect: false,
      };
    }
    case OPEN_ORDER_ITEM: {
      return {
        ...state,
        viewFullOrderModalOpen: true,
        viewFullOrder: action.item,
      };
    }
    case CLOSE_ORDER_ITEM: {
      return {
        ...state,
        viewFullOrderModalOpen: false,
      };
    }
    default: {
      return state;
    }
  }
};
