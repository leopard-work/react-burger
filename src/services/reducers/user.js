import Cookies from 'js-cookie';

import {
    GET_REGISTER_REQUEST,
    GET_REGISTER_SUCCESS,
    GET_REGISTER_FAILED,
    GET_LOGIN_REQUEST,
    GET_LOGIN_SUCCESS,
    GET_LOGIN_FAILED,
    GET_TOKEN_FAILED,
    GET_TOKEN_SUCCESS,
    GET_TOKEN_REQUEST,
    GET_USERINFO_FAILED,
    GET_USERINFO_SUCCESS,
    GET_USERINFO_REQUEST
} from "../actions/user";

const initialState = {
    user: '',
    refreshToken: '',
    accessToken: '',
    registerRequest: false,
    registerFailed: false,
    registerSuccess: false,
    loginRequest: false,
    loginFailed: false,
    loginSuccess: false,
    tokenRequest: false,
    tokenFailed: false,
    tokenSuccess: false,
    userInfoRequest: false,
    userInfoFailed: false,
    userInfoSuccess: false
};

export const userReducer = (state = initialState, action) => {
    switch (action.type) {
        case GET_REGISTER_REQUEST: {
            return {
                ...state,
                registerRequest: true
            };
        }
        case GET_REGISTER_SUCCESS: {
            Cookies.set('token', action.data.refreshToken, { expires: 7, path: '/' });
            return {
                ...state,
                user: action.data.user,
                accessToken: action.data.accessToken.split('Bearer ')[1],
                refreshToken: action.data.refreshToken,
                registerRequest: false,
                registerFailed: false,
                registerSuccess: true
            }
        }
        case GET_REGISTER_FAILED: {
            return {
                ...state,
                registerFailed: true,
                registerRequest: false
            }
        }
        case GET_LOGIN_REQUEST: {
            return {
                ...state,
                loginRequest: true
            };
        }
        case GET_LOGIN_SUCCESS: {
            Cookies.set('token', action.data.refreshToken, { expires: 7, path: '/' });
            return {
                ...state,
                user: action.data.user,
                accessToken: action.data.accessToken.split('Bearer ')[1],
                refreshToken: action.data.refreshToken,
                loginRequest: false,
                loginFailed: false,
                loginSuccess: true
            }
        }
        case GET_LOGIN_FAILED: {
            return {
                ...state,
                loginFailed: true,
                loginRequest: false
            }
        }
        case GET_TOKEN_REQUEST: {
            return {
                ...state,
                tokenRequest: true
            };
        }
        case GET_TOKEN_SUCCESS: {
            Cookies.set('token', action.data.refreshToken, { expires: 7, path: '/' });
            return {
                ...state,
                accessToken: action.data.accessToken.split('Bearer ')[1],
                refreshToken: action.data.refreshToken,
                tokenRequest: false,
                tokenFailed: false,
                tokenSuccess: true
            }
        }
        case GET_TOKEN_FAILED: {
            return {
                ...state,
                tokenFailed: true,
                tokenRequest: false
            }
        }
        case GET_USERINFO_REQUEST: {
            console.log('x');
            return {
                ...state,
                userInfoRequest: true
            };
        }
        case GET_USERINFO_SUCCESS: {
            return {
                ...state,
                user: action.data.user,
                userInfoRequest: false,
                userInfoFailed: false,
                userInfoSuccess: true
            }
        }
        case GET_USERINFO_FAILED: {
            return {
                ...state,
                userInfoFailed: true,
                userInfoRequest: false
            }
        }
        default: {
            return state;
        }
    }
}