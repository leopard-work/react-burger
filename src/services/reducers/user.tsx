import Cookies from "js-cookie";

import {
  GET_REGISTER_REQUEST,
  GET_REGISTER_SUCCESS,
  GET_REGISTER_FAILED,
  GET_LOGIN_REQUEST,
  GET_LOGIN_SUCCESS,
  GET_LOGIN_FAILED,
  GET_TOKEN_FAILED,
  GET_TOKEN_SUCCESS,
  GET_TOKEN_REQUEST,
  GET_USERINFO_FAILED,
  GET_USERINFO_SUCCESS,
  GET_USERINFO_REQUEST,
  GET_UPDATEUSER_REQUEST,
  GET_UPDATEUSER_SUCCESS,
  GET_UPDATEUSER_FAILED,
  GET_LOGOUT_REQUEST,
  GET_LOGOUT_SUCCESS,
  GET_LOGOUT_FAILED,
  GET_FORGOT_REQUEST,
  GET_FORGOT_SUCCESS,
  GET_FORGOT_FAILED,
  SET_FORGOT_EMAIL,
  GET_RESET_REQUEST,
  GET_RESET_SUCCESS,
  GET_RESET_FAILED,
  GET_USER_ALL_ACTIONS,
} from "../actions/user";

type userState = {
  user: string;
  refreshToken: string;
  accessToken: string;
  registerRequest: boolean;
  registerFailed: boolean;
  registerSuccess: boolean;
  loginRequest: boolean;
  loginFailed: boolean;
  loginSuccess: boolean;
  tokenRequest: boolean;
  tokenFailed: boolean;
  tokenSuccess: boolean;
  userCheck: boolean;
  userInfoRequest: boolean;
  userInfoFailed: boolean;
  userInfoSuccess: boolean;
  updateUserRequest: boolean;
  updateUserFailed: boolean;
  updateUserSuccess: boolean;
  logoutRequest: boolean;
  logoutFailed: boolean;
  logoutSuccess: boolean;
  forgotRequest: boolean;
  forgotFailed: boolean;
  forgotSuccess: boolean;
  forgotEmail: string;
  resetRequest: boolean;
  resetFailed: boolean;
  resetSuccess: boolean;
};

export const initialState: userState = {
  user: "",
  refreshToken: "",
  accessToken: "",
  registerRequest: false,
  registerFailed: false,
  registerSuccess: false,
  loginRequest: false,
  loginFailed: false,
  loginSuccess: false,
  tokenRequest: false,
  tokenFailed: false,
  tokenSuccess: false,
  userCheck: true,
  userInfoRequest: false,
  userInfoFailed: false,
  userInfoSuccess: false,
  updateUserRequest: false,
  updateUserFailed: false,
  updateUserSuccess: false,
  logoutRequest: false,
  logoutFailed: false,
  logoutSuccess: false,
  forgotRequest: false,
  forgotFailed: false,
  forgotSuccess: false,
  forgotEmail: "",
  resetRequest: false,
  resetFailed: false,
  resetSuccess: false,
};

export const userReducer = (
  state = initialState,
  action: GET_USER_ALL_ACTIONS
) => {
  switch (action.type) {
    case GET_REGISTER_REQUEST: {
      return {
        ...state,
        registerRequest: true,
      };
    }
    case GET_REGISTER_SUCCESS: {
      Cookies.set("token", action.data.refreshToken, { expires: 7, path: "/" });
      return {
        ...state,
        user: action.data.user,
        accessToken: action.data.accessToken.split("Bearer ")[1],
        refreshToken: action.data.refreshToken,
        registerRequest: false,
        registerFailed: false,
        registerSuccess: true,
      };
    }
    case GET_REGISTER_FAILED: {
      return {
        ...state,
        registerFailed: true,
        registerRequest: false,
      };
    }
    case GET_LOGIN_REQUEST: {
      return {
        ...state,
        loginRequest: true,
      };
    }
    case GET_LOGIN_SUCCESS: {
      Cookies.set("token", action.data.refreshToken, { expires: 7, path: "/" });
      return {
        ...state,
        user: action.data.user,
        accessToken: action.data.accessToken.split("Bearer ")[1],
        refreshToken: action.data.refreshToken,
        loginRequest: false,
        loginFailed: false,
        loginSuccess: true,
        resetSuccess: false,
      };
    }
    case GET_LOGIN_FAILED: {
      return {
        ...state,
        loginFailed: true,
        loginRequest: false,
      };
    }
    case GET_TOKEN_REQUEST: {
      return {
        ...state,
        tokenRequest: true,
      };
    }
    case GET_TOKEN_SUCCESS: {
      Cookies.set("token", action.data.refreshToken, { expires: 7, path: "/" });
      return {
        ...state,
        accessToken: action.data.accessToken.split("Bearer ")[1],
        refreshToken: action.data.refreshToken,
        tokenRequest: false,
        tokenFailed: false,
        tokenSuccess: true,
        userCheck: false,
      };
    }
    case GET_TOKEN_FAILED: {
      return {
        ...state,
        tokenFailed: true,
        tokenRequest: false,
        userCheck: false,
      };
    }
    case GET_USERINFO_REQUEST: {
      return {
        ...state,
        userInfoRequest: true,
      };
    }
    case GET_USERINFO_SUCCESS: {
      return {
        ...state,
        user: action.data.user,
        userInfoRequest: false,
        userInfoFailed: false,
        userInfoSuccess: true,
      };
    }
    case GET_USERINFO_FAILED: {
      return {
        ...state,
        userInfoFailed: true,
        userInfoRequest: false,
      };
    }
    case GET_UPDATEUSER_REQUEST: {
      return {
        ...state,
        updateUserRequest: true,
      };
    }
    case GET_UPDATEUSER_SUCCESS: {
      return {
        ...state,
        user: action.data.user,
        updateUserRequest: false,
        updateUserFailed: false,
        updateUserSuccess: true,
      };
    }
    case GET_UPDATEUSER_FAILED: {
      return {
        ...state,
        updateUserFailed: true,
        updateUserRequest: false,
      };
    }
    case GET_LOGOUT_REQUEST: {
      Cookies.remove("token");
      return initialState;
    }
    case GET_LOGOUT_SUCCESS: {
      return {
        ...initialState,
        userCheck: false,
      };
    }
    case GET_LOGOUT_FAILED: {
      return {
        ...state,
        logoutFailed: true,
        logoutRequest: false,
      };
    }
    case GET_FORGOT_REQUEST: {
      return {
        ...state,
        forgotRequest: true,
      };
    }
    case GET_FORGOT_SUCCESS: {
      return {
        ...state,
        forgotRequest: false,
        forgotFailed: false,
        forgotSuccess: true,
      };
    }
    case GET_FORGOT_FAILED: {
      return {
        ...state,
        forgotFailed: true,
        forgotRequest: false,
      };
    }
    case SET_FORGOT_EMAIL: {
      return {
        ...state,
        forgotEmail: action.data,
      };
    }
    case GET_RESET_REQUEST: {
      return {
        ...state,
        resetRequest: true,
      };
    }
    case GET_RESET_SUCCESS: {
      return {
        ...state,
        resetRequest: false,
        resetFailed: false,
        resetSuccess: true,
        forgotSuccess: false,
      };
    }
    case GET_RESET_FAILED: {
      return {
        ...state,
        resetFailed: true,
        resetRequest: false,
      };
    }
    default: {
      return state;
    }
  }
};
