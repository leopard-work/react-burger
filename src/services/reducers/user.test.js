import {
  GET_REGISTER_REQUEST,
  GET_REGISTER_SUCCESS,
  GET_REGISTER_FAILED,
  GET_LOGIN_REQUEST,
  GET_LOGIN_SUCCESS,
  GET_LOGIN_FAILED,
  GET_TOKEN_FAILED,
  GET_TOKEN_SUCCESS,
  GET_TOKEN_REQUEST,
  GET_USERINFO_FAILED,
  GET_USERINFO_SUCCESS,
  GET_USERINFO_REQUEST,
  GET_UPDATEUSER_REQUEST,
  GET_UPDATEUSER_SUCCESS,
  GET_UPDATEUSER_FAILED,
  GET_LOGOUT_REQUEST,
  GET_LOGOUT_SUCCESS,
  GET_LOGOUT_FAILED,
  GET_FORGOT_REQUEST,
  GET_FORGOT_SUCCESS,
  GET_FORGOT_FAILED,
  SET_FORGOT_EMAIL,
  GET_RESET_REQUEST,
  GET_RESET_SUCCESS,
  GET_RESET_FAILED,
  GET_USER_ALL_ACTIONS,
} from "../actions/user";

import { initialState, userReducer } from "./user";
import renderer from "react-test-renderer";
import React from "react";
import { ordersReducer } from "./orders";

const testData = {
  user: {
    email: "test@test.test",
    name: "test",
  },
  accessToken: "12345",
  refreshToken: "12345",
};

describe("Работа с пользователем", () => {
  it("- исходное состояние", () => {
    expect(userReducer(initialState, {})).toEqual(initialState);
    const tree = renderer.create(<>Исходное состояние</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_REGISTER_REQUEST", () => {
    expect(
      userReducer(initialState, {
        type: GET_REGISTER_REQUEST,
      })
    ).toEqual({
      ...initialState,
      registerRequest: true,
    });
    const tree = renderer.create(<>Регистрация запрос</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_REGISTER_SUCCESS", () => {
    expect(
      userReducer(initialState, {
        type: GET_REGISTER_SUCCESS,
        data: testData,
      })
    ).toEqual({
      ...initialState,
      user: testData.user,
      accessToken: undefined,
      refreshToken: testData.refreshToken,
      registerRequest: false,
      registerFailed: false,
      registerSuccess: true,
    });
    const tree = renderer.create(<>Регистрация завершена</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_REGISTER_FAILED", () => {
    expect(
      userReducer(initialState, {
        type: GET_REGISTER_FAILED,
      })
    ).toEqual({
      ...initialState,
      registerFailed: true,
      registerRequest: false,
    });
    const tree = renderer.create(<>Регистрация ошибка</>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("- обработка GET_LOGIN_REQUEST", () => {
    expect(
      userReducer(initialState, {
        type: GET_LOGIN_REQUEST,
      })
    ).toEqual({
      ...initialState,
      loginRequest: true,
    });
    const tree = renderer.create(<>Авторизация запрос</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_LOGIN_SUCCESS", () => {
    expect(
      userReducer(initialState, {
        type: GET_LOGIN_SUCCESS,
        data: testData,
      })
    ).toEqual({
      ...initialState,
      user: testData.user,
      accessToken: undefined,
      refreshToken: testData.refreshToken,
      loginRequest: false,
      loginFailed: false,
      loginSuccess: true,
      resetSuccess: false,
    });
    const tree = renderer.create(<>Авторизация завершена</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_LOGIN_FAILED", () => {
    expect(
      userReducer(initialState, {
        type: GET_LOGIN_FAILED,
      })
    ).toEqual({
      ...initialState,
      loginFailed: true,
      loginRequest: false,
    });
    const tree = renderer.create(<>Авторизация ошибка</>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("- обработка GET_TOKEN_REQUEST", () => {
    expect(
      userReducer(initialState, {
        type: GET_TOKEN_REQUEST,
      })
    ).toEqual({
      ...initialState,
      tokenRequest: true,
    });
    const tree = renderer.create(<>Получение токена запрос</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_TOKEN_SUCCESS", () => {
    expect(
      userReducer(initialState, {
        type: GET_TOKEN_SUCCESS,
        data: testData,
      })
    ).toEqual({
      ...initialState,
      accessToken: undefined,
      refreshToken: testData.refreshToken,
      tokenRequest: false,
      tokenFailed: false,
      tokenSuccess: true,
      userCheck: false,
    });
    const tree = renderer.create(<>Получение токена завершено</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_TOKEN_FAILED", () => {
    expect(
      userReducer(initialState, {
        type: GET_TOKEN_FAILED,
      })
    ).toEqual({
      ...initialState,
      tokenFailed: true,
      tokenRequest: false,
      userCheck: false,
    });
    const tree = renderer.create(<>Получение токена ошибка</>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("- обработка GET_USERINFO_REQUEST", () => {
    expect(
      userReducer(initialState, {
        type: GET_USERINFO_REQUEST,
      })
    ).toEqual({
      ...initialState,
      userInfoRequest: true,
    });
    const tree = renderer.create(<>Получение информации запрос</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_USERINFO_SUCCESS", () => {
    expect(
      userReducer(initialState, {
        type: GET_USERINFO_SUCCESS,
        data: testData,
      })
    ).toEqual({
      ...initialState,
      user: testData.user,
      userInfoRequest: false,
      userInfoFailed: false,
      userInfoSuccess: true,
    });
    const tree = renderer.create(<>Получение информации завершено</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_USERINFO_FAILED", () => {
    expect(
      userReducer(initialState, {
        type: GET_USERINFO_FAILED,
      })
    ).toEqual({
      ...initialState,
      userInfoFailed: true,
      userInfoRequest: false,
    });
    const tree = renderer.create(<>Получение информации ошибка</>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("- обработка GET_UPDATEUSER_REQUEST", () => {
    expect(
      userReducer(initialState, {
        type: GET_UPDATEUSER_REQUEST,
      })
    ).toEqual({
      ...initialState,
      updateUserRequest: true,
    });
    const tree = renderer.create(<>Обновление информации запрос</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_UPDATEUSER_SUCCESS", () => {
    expect(
      userReducer(initialState, {
        type: GET_UPDATEUSER_SUCCESS,
        data: testData,
      })
    ).toEqual({
      ...initialState,
      user: testData.user,
      updateUserRequest: false,
      updateUserFailed: false,
      updateUserSuccess: true,
    });
    const tree = renderer.create(<>Обновление информации завершено</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_UPDATEUSER_FAILED", () => {
    expect(
      userReducer(initialState, {
        type: GET_UPDATEUSER_FAILED,
      })
    ).toEqual({
      ...initialState,
      updateUserFailed: true,
      updateUserRequest: false,
    });
    const tree = renderer.create(<>Обновление информации ошибка</>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("- обработка GET_LOGOUT_REQUEST", () => {
    expect(
      userReducer(initialState, {
        type: GET_LOGOUT_REQUEST,
      })
    ).toEqual({
      ...initialState,
    });
    const tree = renderer.create(<>Выход запрос</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_LOGOUT_SUCCESS", () => {
    expect(
      userReducer(initialState, {
        type: GET_LOGOUT_SUCCESS,
      })
    ).toEqual({
      ...initialState,
      userCheck: false,
    });
    const tree = renderer.create(<>Выход успешно</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_LOGOUT_FAILED", () => {
    expect(
      userReducer(initialState, {
        type: GET_LOGOUT_FAILED,
      })
    ).toEqual({
      ...initialState,
      logoutFailed: true,
      logoutRequest: false,
    });
    const tree = renderer.create(<>Выход ошибка</>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("- обработка GET_FORGOT_REQUEST", () => {
    expect(
      userReducer(initialState, {
        type: GET_FORGOT_REQUEST,
      })
    ).toEqual({
      ...initialState,
      forgotRequest: true,
    });
    const tree = renderer.create(<>Забыли пароль запрос</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_FORGOT_SUCCESS", () => {
    expect(
      userReducer(initialState, {
        type: GET_FORGOT_SUCCESS,
      })
    ).toEqual({
      ...initialState,
      forgotRequest: false,
      forgotFailed: false,
      forgotSuccess: true,
    });
    const tree = renderer.create(<>Забыли пароль успешно</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_FORGOT_FAILED", () => {
    expect(
      userReducer(initialState, {
        type: GET_FORGOT_FAILED,
      })
    ).toEqual({
      ...initialState,
      forgotFailed: true,
      forgotRequest: false,
    });
    const tree = renderer.create(<>Забыли пароль ошибка</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка SET_FORGOT_EMAIL", () => {
    expect(
      userReducer(initialState, {
        type: SET_FORGOT_EMAIL,
        data: testData.user.email,
      })
    ).toEqual({
      ...initialState,
      forgotEmail: testData.user.email,
    });
    const tree = renderer.create(<>Запоминание email</>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("- обработка GET_RESET_REQUEST", () => {
    expect(
      userReducer(initialState, {
        type: GET_RESET_REQUEST,
      })
    ).toEqual({
      ...initialState,
      resetRequest: true,
    });
    const tree = renderer.create(<>Обновление пароля запрос</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_RESET_SUCCESS", () => {
    expect(
      userReducer(initialState, {
        type: GET_RESET_SUCCESS,
      })
    ).toEqual({
      ...initialState,
      resetRequest: false,
      resetFailed: false,
      resetSuccess: true,
      forgotSuccess: false,
    });
    const tree = renderer.create(<>Обновление пароля успешно</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка GET_RESET_FAILED", () => {
    expect(
      userReducer(initialState, {
        type: GET_RESET_FAILED,
      })
    ).toEqual({
      ...initialState,
      resetFailed: true,
      resetRequest: false,
    });
    const tree = renderer.create(<>Обновление пароля ошибка</>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
