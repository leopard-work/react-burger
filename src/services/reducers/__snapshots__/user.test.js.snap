// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Работа с пользователем - исходное состояние 1`] = `"Исходное состояние"`;

exports[`Работа с пользователем - обработка GET_FORGOT_FAILED 1`] = `"Забыли пароль ошибка"`;

exports[`Работа с пользователем - обработка GET_FORGOT_REQUEST 1`] = `"Забыли пароль запрос"`;

exports[`Работа с пользователем - обработка GET_FORGOT_SUCCESS 1`] = `"Забыли пароль успешно"`;

exports[`Работа с пользователем - обработка GET_LOGIN_FAILED 1`] = `"Авторизация ошибка"`;

exports[`Работа с пользователем - обработка GET_LOGIN_REQUEST 1`] = `"Авторизация запрос"`;

exports[`Работа с пользователем - обработка GET_LOGIN_SUCCESS 1`] = `"Авторизация завершена"`;

exports[`Работа с пользователем - обработка GET_LOGOUT_FAILED 1`] = `"Выход ошибка"`;

exports[`Работа с пользователем - обработка GET_LOGOUT_REQUEST 1`] = `"Выход запрос"`;

exports[`Работа с пользователем - обработка GET_LOGOUT_SUCCESS 1`] = `"Выход успешно"`;

exports[`Работа с пользователем - обработка GET_REGISTER_FAILED 1`] = `"Регистрация ошибка"`;

exports[`Работа с пользователем - обработка GET_REGISTER_REQUEST 1`] = `"Регистрация запрос"`;

exports[`Работа с пользователем - обработка GET_REGISTER_SUCCESS 1`] = `"Регистрация завершена"`;

exports[`Работа с пользователем - обработка GET_RESET_FAILED 1`] = `"Обновление пароля ошибка"`;

exports[`Работа с пользователем - обработка GET_RESET_REQUEST 1`] = `"Обновление пароля запрос"`;

exports[`Работа с пользователем - обработка GET_RESET_SUCCESS 1`] = `"Обновление пароля успешно"`;

exports[`Работа с пользователем - обработка GET_TOKEN_FAILED 1`] = `"Получение токена ошибка"`;

exports[`Работа с пользователем - обработка GET_TOKEN_REQUEST 1`] = `"Получение токена запрос"`;

exports[`Работа с пользователем - обработка GET_TOKEN_SUCCESS 1`] = `"Получение токена завершено"`;

exports[`Работа с пользователем - обработка GET_UPDATEUSER_FAILED 1`] = `"Обновление информации ошибка"`;

exports[`Работа с пользователем - обработка GET_UPDATEUSER_REQUEST 1`] = `"Обновление информации запрос"`;

exports[`Работа с пользователем - обработка GET_UPDATEUSER_SUCCESS 1`] = `"Обновление информации завершено"`;

exports[`Работа с пользователем - обработка GET_USERINFO_FAILED 1`] = `"Получение информации ошибка"`;

exports[`Работа с пользователем - обработка GET_USERINFO_REQUEST 1`] = `"Получение информации запрос"`;

exports[`Работа с пользователем - обработка GET_USERINFO_SUCCESS 1`] = `"Получение информации завершено"`;

exports[`Работа с пользователем - обработка SET_FORGOT_EMAIL 1`] = `"Запоминание email"`;
